open-hackathon-adminUI/src/app/database/models.py

```diff
@@ -0,0 +1,161 @@
+
+    def get_admin_user_id(self):
+        return self.id
+
+    def json(self):
+        return to_json(self, self.__class__)
+
+    def __init__(self, **kwargs):
+        super(AdminUser, self).__init__(**kwargs)
+        if self.create_time is None:
+            self.create_time = datetime.utcnow()
+        if self.last_login_time is None:
+            self.last_login_time = datetime.utcnow()
+#        if self.slug is None:
+#            self.slug = str(uuid.uuid1())[0:8]  # todo generate a real slug
+
+    def __repr__(self):
+        return "AdminUser: " + self.json()
+
+    def is_authenticated(self):
+        return True
+
+    def is_active(self):
+        return True
+
+    def is_anonymous(self):
+        return False
+
+    def get_id(self):
+        return unicode(self.id)
+
+
+
+class AdminEmail(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(80))
+    email = db.Column(db.String(120))
+    primary_email = db.Column(db.Integer)  # 0:NOT Primary Email 1:Primary Email
+    verified = db.Column(db.Integer)  # 0 for not verified, 1 for verified
+
+    admin_id = db.Column(db.Integer, db.ForeignKey('admin_user.id', ondelete='CASCADE'))
+    admin = db.relationship('AdminUser', backref=db.backref('emails', lazy='dynamic'))
+
+    def get_admin_email(self):
+        return self.email(self, email)
+
+    def __init__(self, **kwargs):
+        super(AdminEmail, self).__init__(**kwargs)
+
+
+
+class AdminToken(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    token = db.Column(db.String(50), unique=True, nullable=False)
+
+    admin_id = db.Column(db.Integer, db.ForeignKey('admin_user.id', ondelete='CASCADE'))
+    admin = db.relationship('AdminUser', backref=db.backref('tokens', lazy='dynamic'))
+
+    issue_date = db.Column(db.DateTime)
+    expire_date = db.Column(db.DateTime, nullable=False)
+
+    def json(self):
+        return to_json(self, self.__class__)
+
+    def __init__(self, token, admin, expire_date, issue_date=None):
+        if issue_date is None:
+            issue_date = datetime.utcnow()
+
+        self.token = token
+        self.admin = admin
+        self.expire_date = expire_date
+        self.issue_date = issue_date
+
+    def __repr__(self):
+        return "AdminToken: " + self.json()
+
+
+
+class AdminGroup(db.Model):
+
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(16))
+    hackathon_id = db.Column(db.Integer)
+    create_time = db.Column(db.DateTime)
+
+    def json(self):
+        return to_json(self, self.__class__)
+
+    def __init__(self, **kwargs):
+        super(AdminGroup, self).__init__(**kwargs)
+
+    def __repr__(self):
+        return "AdminGroup: " + self.json()
+
+
+
+class AdminUserGroup(db.Model):
+
+    id = db.Column(db.Integer, primary_key=True)
+    group_id = db.Column(db.Integer, unique=True, nullable=False)
+    admin_email = db.Column(db.String(120))
+    state = db.Column(db.String(16))
```

Comment:
sometime we will disable the management authority of admin to hackathons , this is meaning of the 'state'  And change the type to int(2)